### PR TITLE
update support bundle to include new data gathering

### PR DESCRIPTION
**Pull request by dthomson-sysdig**
- /api/admin/customer/1/dataRetentionSettings
- metrics (syscall.count, dragent.analyzer.sr, container.count, dragent.analyzer.n_drops_buffer, dragent.analyzer.n_evts)
-  Get Agent Versions and Metric Limits data from the API to see how many metrics agents are indeed sending.
- cleaned up some variables to standardize on formatting (eg; $pod -> ${pod})
- Added mysql storage info (if it exists)
- Added ability to determine if container has **sh** or **bash** installed when we try collect logs using **tar** inside the container.
- Standardize on storage location for checks we are doing against the statefulsets.
- Add snapshotSettings API Call
- Add ElasticSearch SSL Certificate End Date Data Collection
- Add api/license Call if customer passes in API_KEY
- Refactor argument parsing 
- Add --skip-logs optional arg so we don't collect any logs from kubectl or the pods/containers
- Add "get storageclass" kubectl command
- Add "get nodes" JSON via kubectl

### CODE REVIEW

File: support_bundle/get_support_bundle.sh [hunk 1]
```diff
@@ -11,216 +11,396 @@ catch() {
...
+    fi
+    
+    # Configure kubectl command if labels are set
+    if [[ -z ${LABELS} ]]; then
+        SYSDIGCLOUD_PODS=$(kubectl ${KUBE_OPTS} get pods | awk '{ print $1 }' | grep -v NAME)
+    else
+        SYSDIGCLOUD_PODS=$(kubectl ${KUBE_OPTS} -l "role in (${LABELS})" get pods | awk '{ print $1 }' | grep -v NAME)
+    fi
+    
+    echo "Using namespace ${NAMESPACE}";
```
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:211]: nit, but please remove trailing `;`

Comment (dthomson-sysdig) [reply support_bundle/get_support_bundle.sh]: @mbreitung fixed in new commit [https://github.com/draios/sysdig-cloud-scripts/pull/191/commits/2e60d77bf2a5c914ce5ec79abe1bbb7d57a0c984](https://github.com/draios/sysdig-cloud-scripts/pull/191/commits/2e60d77bf2a5c914ce5ec79abe1bbb7d57a0c984)

File: support_bundle/get_support_bundle.sh [hunk 2]
```diff
@@ -11,216 +11,367 @@ catch() {
...
+                ;;
+            -c|--context)
+                test $# -lt 2 && die "Missing value for the optional argument '$_key'." 1
+                CONTEXT="$2"
+                shift
+                ;;
+            -d|--debug)
+                set -x
+                ;;
+            -d*)
```
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:52]: hah, nice

File: support_bundle/get_support_bundle.sh [hunk 3]
```diff
@@ -11,216 +11,367 @@ catch() {
...
+    if [[ "$(echo "$KUBE_OUTPUT" | grep -o "^sysdig " || true)" != "${NAMESPACE} " ]]; then
+        echo "We could not determine the namespace. Please check the spelling and try again";
+        echo "kubectl ${KUBE_OPTS} get ns";
+        echo "$(kubectl ${KUBE_OPTS} get ns)";
+    fi
+    # If API key is supplied, collect streamSnap, Index settings, and fastPath settings
+    if [[ ! -z ${API_KEY} ]]; then
+        API_URL=$(kubectl ${KUBE_OPTS} get cm sysdigcloud-config -o yaml | grep -i api.url: | head -1 | awk '{print$2}')
+        curl -ks -H "Authorization: Bearer ${API_KEY}" -H "Content-Type: application/json" "${API_URL}/api/license" >> ${LOG_DIR}/license.json
+        curl -ks -H "Authorization: Bearer ${API_KEY}" -H "Content-Type: application/json" "${API_URL}/api/admin/customer/1/streamsnapSettings" >> ${LOG_DIR}/streamSnap_settings.json
```
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:151]: this is fine for now, but note that we have had customers in the past use a different customer id than `1`

File: support_bundle/get_support_bundle.sh [hunk 4]
```diff
@@ -11,216 +11,367 @@ catch() {
...
+    echo "Collecting node information"
+    kubectl ${KUBE_OPTS} describe nodes | tee -a ${LOG_DIR}/describe_node_output.log || echo "No permission to describe nodes!"
+
+    NODES=$(kubectl ${KUBE_OPTS} get nodes --no-headers | awk '{print $1}') && RETVAL=0 || { RETVAL=$? && echo "No permission to get nodes!"; }
+    if [[ "${RETVAL}" == "0" ]]; then
+        mkdir -p ${LOG_DIR}/nodes
+        for node in ${NODES[@]}; do
+            kubectl ${KUBE_OPTS} get node ${node} -ojson > ${LOG_DIR}/nodes/${node}-kubectl.json
+        done
+        unset RETVAL
```
Comment (mbreitung) [inline support_bundle/get_support_bundle.sh:231]: should this be outside of the `if` block?

**Review by mbreitung — APPROVED** [review]

File: support_bundle/get_support_bundle.sh (hunk 4 above)
Comment (dthomson-sysdig) [reply support_bundle/get_support_bundle.sh:231]: Yes, if they choose to skip logs, we still want to get the json description of the pod (small amount of data) and allows us to still analyze the if needed.  That's why I moved it outside of the log block, as technically it isn't a "log"

**Review by patrickhargett — APPROVED** [review]
LGTM! Nice work Drew!

**Pull request merged**